src/JuliaFormatter.jl

```diff
@@ -549,6 +552,23 @@ function format_text(text::AbstractString, style::AbstractStyle; kwargs...)
     return format_text(text, style, opts)
 end
 
+function format_text(text::AbstractString, style::SciMLStyle; maxiters = 3, kwargs...)
+    isempty(text) && return text
+    opts = Options(; merge(options(style), kwargs)...)
+    # We need to iterate to a fixpoint because the result of short to long
+    # form isn't properly formatted
+    formatted_text = format_text(text, style, opts)
+    iter = 1
+    while formatted_text != text
+        iter > maxiters &&
+            error("formatted_text hasn't reached to a fixpoint in $iter iterations")
+        text = formatted_text
+        formatted_text = format_text(text, style, opts)
+        iter += 1
+    end
+    return formatted_text
+end
```

Comment:
That feels quite hacky? Also, not clear how this would interact with a potential partial formatting in an IDE? Wouldn't it make sense to require that a second call to `format_text` is always a no-op?